float32 ndist
int32 next_index
---
int32 success